### DISCUSSION TITLE
Dynamic/programmatic Resources

### DISCUSSION THREAD

**Discussion by ioson** [filamentphp/filament#8697] [Help] [answered]
Hi everyone,
I'm currently trying to build a reusable Backend/CMS with filament.
Is there any way to programmatically define Resources?

The Backend would have a page where the user can create resource types. For example: Bars, Clubs and Restaurants.
Each of these types should be listed in the Backend navigation. They would be stored as the same model, having a “type” column to differentiate them. (Basically like WordPress does it with their (custom) post types).

I already extended the Resource class to share the same form and table with all my Resources. But I still need to create a Resource file and a folder with the views for every type.

Any idea how to approach this? Is this something that could be done with filament?

Cheers Jason

**Comment by leandrocfe** [comment, marked as answer]
Yes, you can develop it. I've never used [this plugin](https://filamentphp.com/plugins/lara-zeus-bolt) before, but you can have a look.

**Comment by gixxy22** [comment]
Hi Jason,

Did you have any luck with this functionality? Looking fot exactly the same thing, the Bolt plugin didnt seem to fit the specs for this.

**Reply by ioson** [reply]
Hi, no i didn't find a solution and just use a single filament resource now. I added a filter in the table view to differentiate the types. That's good enough for now.
